src/datasources/perf-ds/PerformanceHelpers.ts

```diff
@@ -9,93 +9,101 @@ import { OnmsMeasurementsQueryResponse } from "./types";
-
-  for (let i = 0; i < timestamps.length; i++) {
-      const tsVal = timestamps[i]
-
-      if (end > 0 && tsVal > end) {
-          endIndex = i - 1
-          break
-      }
-
-      if (start === 0 || tsVal >= start) {
-          if (startIndex < 0) {
-              startIndex = i
-          }
-          windowedTimestamps.push(tsVal)
-      }
-  }
-
-  return {
-    windowedTimestamps,
-    startIndex,
-    endIndex
-  }
 }
 
 /**
  * Convert QueryResponse data returned by OpenNMS Measurements Rest API to Grafana DataFrame format.
  */
 export const measurementResponseToDataFrame =
-    (measurementData: OnmsMeasurementsQueryResponse, queryRefId?: string): DataFrame => {
+    (measurementData: OnmsMeasurementsQueryResponse): DataFrame[] => {
 
-    const { start, end, labels, columns, timestamps, metadata } = measurementData
+        const { start, end, labels, columns, timestamps, metadata } = measurementData
 
-    const dataFrame: DataFrame = {
-        name: '',
-        refId: queryRefId,
-        length: 0,
-        fields: []
-    }
+        const dataFrames: DataFrame[] = []
 
-    if (!columns || !columns.length || !timestamps || !timestamps.length) {
-        return dataFrame
-    }
+        let dataFrame: DataFrame = getEmptyDataFrame()
 
-    const { windowedTimestamps, startIndex, endIndex } = getWindowedTimestamps(timestamps, start, end)
+        if (!columns || !columns.length || !timestamps || !timestamps.length) {
+            dataFrames.push(dataFrame)
+        }
 
-    // no data or no data within the start/end timespan, return an empty DataFrame
-    if (windowedTimestamps.length === 0) {
-        return dataFrame
-    }
+        const { windowedTimestamps, startIndex, endIndex } = getWindowedTimestamps(timestamps, start, end)
+
+        // no data or no data within the start/end timespan, return an empty DataFrame
+        if (windowedTimestamps.length === 0) {
+            dataFrames.push(dataFrame)
+        }
+
+        for (let i = 0; i < labels?.length; i++) {
+            dataFrame = getEmptyDataFrame()
+            
+            dataFrame.length = windowedTimestamps.length
 
-    dataFrame.length = windowedTimestamps.length
+            dataFrame.fields.push({
+                name: 'Time',
+                type: FieldType.time,
+                config: {},
+                values: new ArrayVector<number>(windowedTimestamps)
+            } as Field)
 
-    dataFrame.fields.push({
-        name: 'Time',
-        type: FieldType.time,
-        config: {},
-        values: new ArrayVector<number>(windowedTimestamps)
-    } as Field)
+            const label = metadata && metadata.resources ?
+                FunctionFormatter.format(labels[i], metadata) :
+                labels[i]
 
-    for (let i = 0; i < columns?.length; i++) {
-        let label = metadata && metadata.resources ?
-            FunctionFormatter.format(labels[i], metadata) :
-            labels[i];
+            const column = columns[i]
```

Comment:
If the checks above are omitted, will need to check that `columns[i]` exists here

Reply:
I'm not sure if the rest response will ever retrun missing properties like columns, labels or timestamps, so kept the checks and skipped processing in case they didn't exist, starting for timestamps then columns. Also I'm not sure how grafana would handle dataFrames with timestamps and no values associated.
So currently 
-no timestamps -> dataframe[] with one empty dataframe
-timestamps and no columns -> dataframe[] with one dataframe with Time only
-timestamps and columns -> dataframe[] with #labels/#columns dataframe with Time plus #columns values